oteapi/plugins/entry_points.py

```diff
@@ -0,0 +1,505 @@
+    ) -> EntryPointStrategy:
+        """Retrieve an entry point from the collection.
+
+        Raises:
+            KeyError: If an entry point cannot be found in the collection.
+
+        Returns:
+            An entry point in the collection representing the key.
+
+        """
+        if isinstance(key, (EntryPointStrategy, str, tuple)):
+            if key not in self:
+                raise KeyError(f"{key} not found in {self}")
+            return self._get_entry_point(key)
+        raise TypeError(
+            "key should either be of type EntryPointStrategy, a string of the full "
+            "name or a strategy tuple."
+        )
+
+    def _get_entry_point(
+        self,
+        key: "Union[EntryPointStrategy, str, Tuple[Union[StrategyType, str], str]]",
+    ) -> EntryPointStrategy:
+        """Helper method for retrieving an entry point from the collection.
+
+        Important:
+            It is expected that the entry point representing the key exists in the
+            collection. For example through a `key in self` test.
+
+        Raises:
+            RuntimeError: If an entry point cannot be found in the collection, since
+                this is a requirement.
+
+        Returns:
+            An entry point in the collection representing the key.
+
+        """
+        if isinstance(key, EntryPointStrategy):
+            return key
+        if isinstance(key, str):
+            for entry_point in self._entry_points:
+                if key == entry_point.full_name:
+                    return entry_point
+        if isinstance(key, tuple):
+            key_ = (StrategyType(key[0]), key[1])
+            for entry_point in self._entry_points:
+                if key_ == entry_point.strategy:
+                    return entry_point
+        raise RuntimeError(
+            f"{key} not found in {self}, which is a requirement for the "
+            "_get_entry_point method."
+        )
+
+    def __eq__(self, other: "Any") -> bool:
+        if isinstance(other, self.__class__):
+            return hash(self) == hash(other)
+        return False
+
+    def __hash__(self) -> int:
+        return hash(tuple(_ for _ in sorted(self._entry_points)))
+
+    def __str__(self) -> str:
+        res = {}  # type: ignore[var-annotated]
+        for _ in self._entry_points:
+            if _.type.value in res:
+                res[_.type.value] += 1
+            else:
+                res[_.type.value] = 1
+        res = sorted(f"{key} ({value})" for key, value in res.items())
+        return f"<{self.__class__.__name__}: " f"Strategies={', '.join(res)}>"
+
+    def __repr__(self) -> str:
+        return f"{self.__class__.__name__}(*{tuple(sorted(self._entry_points))!r})"
+
+
+def get_strategy_entry_points(
+    strategy_type: "Union[StrategyType, str]",
+    enforce_uniqueness: bool = True,
+) -> EntryPointStrategyCollection:
+    """Retrieve all entry points from a specific strategy type.
+
+    Raises:
```

Comment:
Is there a conventional order for the fields in google docstrings? For me it feels more natural that Raises comes after Return. 

This comment applies for other docstrings as well.

Reply:
Okay - for me it was more natural the the last section was `Returns`, and when getting there the reader is already aware what exceptions may occur - but I can switch it up for sure if you want. I think the rendering in the documentation doesn't care either way.

Reply:
It is probably because I have earlier been using [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html), which defines a very specific order of the different section. But I don't think it hurts to use the order they have defined

Reply:
It could be a low priority new issue to not slow down this wonderful PR any further...

Reply:
Good suggestion, as this requires a bit of manual work to go through all the code and cannot be done with a simple search-and-replace. Please make another issue for this.

Reply:
Opened #58